test/pages/productAppExtension/index.spec.tsx

```diff
@@ -0,0 +1,22 @@
+import { useRouter } from "next/router";
+import ProductAppExtension from "@pages/productAppExtension/[productId]"; 
+import { render, screen } from '@test/utils';
+
+jest.mock('@lib/hooks', () => require('@mocks/hooks'));
+jest.mock('next/router', () => ({
+    useRouter: jest.fn(),
+}));
+
+describe('ProductAppExtension', () => {
+    const router = { query: { pid: '1' } };
+    useRouter.mockReturnValue(router);
+
+    test('renders correctly', () => {
+        const { container } = render(<ProductAppExtension />);
+        const panelOneHeader = screen.getByText("Basic Information");
+
+        expect(container.firstChild).toMatchSnapshot();
+        expect(panelOneHeader).toBeInTheDocument();
+    });
+});
+
```

Comment:
🍹 I might suggest testing for error states. For example, what happens if `router.query.pid` is undefined? What should the UI render in that case?